### PR TITLE
*Implement events data for self hosted video content on mobile web

### PR DESCRIPTION
**Pull request by MoMitch**
This implements async event tracking for several consumption events of the new self-hosted video content.

### CODE REVIEW

**Comment by MoMitch** [comment]
@birakattack

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 1]
```diff
@@ -502,6 +550,201 @@ class HTML5StreamPlayer extends React.Component {
...
+    const payload = {
+      ...this.buildBaseEventData(),
+      source: 'videoplayer',
+      verb: 'scroll',
+      noun: 'autoplay',
+    }
+    this.sendTrackVideoEvent('videoplayer__scroll_autoplay',payload);
+  }
+
+  getPctServed() {
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: nit: can we just spell out percent here, so `getPercentServed()`?

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 2]
```diff
@@ -502,6 +550,201 @@ class HTML5StreamPlayer extends React.Component {
...
+
+  sendVideoServedEvent() {
+    const payload = {
+      ...this.buildBaseEventData(),
+      ...this.getPctServed(),
+      source: 'videoplayer',
+      verb: 'served',
+      noun: 'video',
+    }
+    this.sendTrackVideoEvent('videoplayer__served_video',payload);
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: Can we just move the source, verb, noun stuff into the buildBaseEventData method and just pass the `event_type` and parse that to set those values?

Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: In fact can we just move all of this logic into `this.sendTrackVideoEvent` and allow an optional parameter that will be splatted into the base payload?

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 3]
```diff
@@ -502,6 +551,117 @@ class HTML5StreamPlayer extends React.Component {
...
+      sr_name: postData.subreddit,
+      sr_fullname: postData.subredditId,
+      sr_id: parseInt(subredditShortID, 36),
+      page_type: pageType,
+    };
+
+    return payload;
+  }
+
+  sendTrackVideoEvent(eventType) {
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: We should just add an optional parameter to this method that can take an object that should be splatted with the base event payload. 

```
sendTrackVideoEvent(eventType, optionalParams={}) {
    let payload = { 
        ...this.buildBaseEventData(),
        ...optionalParams,
    };
    this.props.dispatch(trackVideoEvent(eventType,payload));
}
```

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 4]
```diff
@@ -502,6 +551,117 @@ class HTML5StreamPlayer extends React.Component {
...
+      max_timestamp_served: parseInt(this.state.totalServedTime),
+      percent_served: pctServed,
+    };
+
+    return payload;
+  }
 }
 
-export default HTML5StreamPlayer;
+export default connect(null)(HTML5StreamPlayer);
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: Why the `connect(null)` here?

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 5]
```diff
@@ -91,10 +95,12 @@ class HTML5StreamPlayer extends React.Component {
       const video = this.refs.HTML5StreamPlayerVideo;
       if (videoIsInView) {
         if (video.paused && !window.fullScreen) {
+          this.sendTrackVideoEvent(VIDEO_EVENT.SCROLL_AUTOPLAY);
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: Should we move this call below the `video.play();` call?

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 6]
```diff
@@ -91,10 +95,12 @@ class HTML5StreamPlayer extends React.Component {
...
           video.play();
         }
       } else {
         if (!video.paused) {
+          this.sendTrackVideoEvent(VIDEO_EVENT.SCROLL_PAUSE);
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: Same as above should we move this to after `video.pause();` is called?

File: src/app/components/HTML5StreamPlayer/index.jsx [hunk 7]
```diff
@@ -502,6 +551,117 @@ class HTML5StreamPlayer extends React.Component {
...
+      target_author_id: parseInt(postData.author, 36),
+      target_author_name: postData.author,
+      target_created_ts: msSinceCreation,
+      target_id: parseInt(postData.id, 36),
+      target_url: postData.cleanUrl,
+      target_url_domain: postData.domain,
+      sr_name: postData.subreddit,
+      sr_fullname: postData.subredditId,
+      sr_id: parseInt(subredditShortID, 36),
+      page_type: pageType,
```
Comment (birakattack) [inline src/app/components/HTML5StreamPlayer/index.jsx]: Can we add a `target_type` as well? It was recently added to the spec but essentially we want to know whether its a video or a gif.

File: src/app/components/HTML5StreamPlayer/index.jsx (hunk 6 above)
Comment (MoMitch) [inline src/app/components/HTML5StreamPlayer/index.jsx]: As above.

File: src/app/components/HTML5StreamPlayer/index.jsx (hunk 5 above)
Comment (MoMitch) [inline src/app/components/HTML5StreamPlayer/index.jsx]: Correct, moving now.

File: src/app/components/HTML5StreamPlayer/index.jsx (hunk 7 above)
Comment (MoMitch) [inline src/app/components/HTML5StreamPlayer/index.jsx]: what should this return?

File: src/app/components/HTML5StreamPlayer/index.jsx (hunk 4 above)
Comment (MoMitch) [inline src/app/components/HTML5StreamPlayer/index.jsx]: changed to connect() instead, but the connect in general is here to allow the getState for analytics.

**Comment by MoMitch** [comment]
:eyeglasses: @birakattack

**Pull request merged**